src/pyspiffe/workloadapi/default_workload_api_client.py

```diff
@@ -42,55 +45,86 @@ def __init__(self, spiffe_socket: str = None) -> None:
             self._config = ConfigSetter(
                 spiffe_endpoint_socket=spiffe_socket
             ).get_config()
-        except ValueError:
+        except ValueError as e:
             raise ValueError(
-                'SPIFFE socket argument or environment variable invalid in DefaultWorkloadApiClient.'
+                'Invalid DefaultWorkloadApiClient configuration: {}'.format(str(e))
             )
 
         self._spiffe_workload_api_stub = workload_pb2_grpc.SpiffeWorkloadAPIStub(
             self._get_spiffe_grpc_channel()
         )
 
     def fetch_x509_svid(self) -> X509Svid:
-        """Fetches a SPIFFE X.509-SVID.
+        """Fetches the default X509-SVID, i.e. the first in the list returned by the Workload API.
 
         Returns:
             X509Svid: Instance of X509Svid object.
         """
```

Comment:
This method will raise FetchX509SvidError in case there is an error connecting to the agent or no item is returned. It can also raise some exceptions originated by X509Svid.parse_raw. These should be added to docstring. The same for the methods below.

Reply:
Added exception handling to all the methods, catching the exceptions from the `parse_raw` methods and re-throwing with more context.